File: MyDailyBeat/MyDailyBeat/EVCRegistrationMessageViewController.m
//
//  EVCRegistrationMessageViewController.m
//  MyDailyBeat
//
//  Created by Virinchi Balabhadrapatruni on 9/19/16.
//  Copyright © 2016 eVerveCorp. All rights reserved.
//

#import "EVCRegistrationMessageViewController.h"
#import "EVCRegistrationViewController.h"

@interface EVCRegistrationMessageViewController ()

@end

@implementation EVCRegistrationMessageViewController

- (id) initWithKey: (int) key {
    self = [super initWithNibName:@"EVCRegistrationMessageViewController" bundle:nil];
    if (self) {
        if (key == 1) {
            self.message = WELCOME_MESSAGE_1;
        } else {
            self.message = WELCOME_MESSAGE_2;
        }
        self.key = key;
        
    }
    return self;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    [messageLabel setText:self.message];
    self.navigationController.navigationBar.tintColor = UIColorFromHex(0x0097A4);
}

- (void) message1 {
    EVCRegistrationMessageViewController *message2 = [[EVCRegistrationMessageViewController alloc] initWithKey:2];
    [self.navigationController pushViewController:message2 animated:YES];
}

- (void) message2 {
    EVCRegistrationViewController *controller = [[EVCRegistrationViewController alloc] init];
    [self.navigationController pushViewController:controller animated:YES];
}

- (IBAction)next:(id)sender {
    if (_key == 1) {
        [self message1];
    } else {
        [self message2];
    }
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: MDB_Swift/MyDailyBeat/MyDailyBeat/src/EVCRegistrationMessageViewController.swift

//
//  EVCRegistrationMessageViewController.swift
//  MyDailyBeat
//
//  Created by Virinchi Balabhadrapatruni on 9/19/16.
//  Copyright © 2016 eVerveCorp. All rights reserved.
//
import UIKit
import API
class EVCRegistrationMessageViewController: UIViewController {

    @IBOutlet var messageLabel: UILabel!
    @IBOutlet var nextButton: UIButton!
    @IBOutlet var imageView: UIImageView!
    var message: String = ""
    var image: UIImage? = nil
    var index = 0
    var nextPage: (() -> ()) = {
        // empty by default
    }

    @IBAction func next(_ sender: Any) {
        self.nextPage()
    }
    
    override var nibName: String? {
        get {
            return "RegistrationMessageViewController"
        }
    }


    override func viewDidLoad() {
        super.viewDidLoad()
        messageLabel.text = self.message
        if let nonNil = self.image {
            self.imageView.image = nonNil
        }
        
    }

    
}
